utils/reactiveApi.ts

```diff
@@ -2,7 +2,11 @@ import type { AsyncData } from "nuxt/app";
 
 export default function reactiveApi<Data, Error>(data: AsyncData<Data, Error>) {
   const stopHandler = watch(
-    [useNuxtApp().$i18n.locale, () => contentLanguageStore().contentLanguages],
+    [
+      useNuxtApp().$i18n.locale,
+      () => contentLanguageStore().contentLanguages,
+      () => contentLanguageStore().selectedLanguage,
```

Comment:
The other option is quite a bit of work: reload from the point where the page called "setQueue" - there we got the context information of reloading it... and we got about 8 locations of the "PlayList" element where the tracks are transmitted as prop. I would prefer to cache the corresponding tracklist URL somewhere to have it available in the TrackChangeLocale component.

I guess the more elegant solution would be to simply load the track by its Id. But shouldn't it affect the current playlist as well?

Reply:
The app does it for the currently playing track only. Not the whole queue.
I think that's actually fine. The only case where it would be nice to apply it to the whole queue, is for an audiobook. But they are currently uploaded as separate tracks.
Summary: Only the current track is fine.

Reply:
Now only the currently playing track should be reloaded. Note that I had to assign the queue index to itself to make it reload the track media

Reply:
fixed the queue index issue and I also keep the position now